packages/rpc-spec/src/rpc.ts

```diff
@@ -60,7 +60,7 @@ function makeProxy<TRpcMethods, TRpcTransport extends RpcTransport>(
                 const createRpcRequest = Reflect.get(target, methodName, receiver);
                 const newRequest = createRpcRequest
                     ? createRpcRequest(...rawParams)
-                    : { methodName, params: rawParams };
+                    : { payload: createRpcMessage(methodName, rawParams) };
```

Comment:
There's a little caveat here.

Since it is possible that `Reflect.get` will return undefined when trying to access the request plan, the outer RPC layer needs to defined a "fallback plan" which is why, in some situations, it may end up using `createRpcMessage`.

An alternative would be to throw an error if no plan can be constructed from the given API.

Reply:
I think I'd prefer to have a solana error thrown if `createRpcRequest` is undefined after the `Reflect.get` 

I'm guessing this would happen if you eg called `rpc.methodThatDoesntExist()`?

Reply:
This would only happen if you provided an RPC API that explicitly defined all the supported methods but none of the others. Our RPC API does not do this, it uses a proxy object to catch any method call. This means, with our default RPC API, this "fallback" scenario is never used. Which makes me think it's probably a good thing to throw if a plan could not be created.

Reply:
I actually did exactly this in the subscriptions API that I'm refactoring, because it makes zero sense to have a ‘fallback’ strategy for making a subscription. Like, what would that even be – subscription protocols are all so specific.

Reply:
Thanks guys! I've made the change in [a follow-up PR](https://github.com/solana-labs/solana-web3.js/pull/3245).